### ISSUE TITLE
[Bug] Unity rendering beneath flutter on Android 7 device.

### ISSUE THREAD

**Issue by timbotimbo** [juicycleff/flutter-unity-view-widget#317]
**Describe the bug**
The flutter-unity-widget 3.0.0 example works fine on most devices I tested with, but it runs into an issue on my budget android 7 testing device. I don't have any other devices running old android versions like this, so I don't know if it is simply this phone or all phones on this android version.

Unity does not appear to render, showing the menuScreen widget in the background.
On further inspection, Unity is running but it is behind the flutter widgets.

After clicking Simple Unity demo, unity is invisible but logs show onSceneLoaded events.
menuScreen before and after clicking Simple Unity Demo:
![Screenshot_20210210-175110-small](https://user-images.githubusercontent.com/11444698/107545861-0cf7d100-6bcc-11eb-9301-209592368b8a.png)  ![Screenshot_20210210-175119-small](https://user-images.githubusercontent.com/11444698/107545988-2d279000-6bcc-11eb-8429-c7f120444919.png)

The orientation test allows me to see the scene for a split second in landscape mode.
![Screenshot_20210210-175210-small](https://user-images.githubusercontent.com/11444698/107547807-28fc7200-6bce-11eb-8f2e-d5d47394df42.png)  ![Screenshot_20210210-175135-small](https://user-images.githubusercontent.com/11444698/107547820-2c8ff900-6bce-11eb-9c13-d45e0f41b630.png)

If I change the Unity player setting to `start in fullscreen mode', the scene also shows up at the bottom.
![Screenshot_20210210-175700-small](https://user-images.githubusercontent.com/11444698/107546739-fdc55300-6bcc-11eb-8aa6-62e951c57180.png)

Deploying the outdated AR samples project with `flutter_unity_widget: ^0.1.6+5`, does show unity on top.
(This phone doesn't have ARcore installed, that is why there is no visible camera image)
![Screenshot_20210210-184643-small](https://user-images.githubusercontent.com/11444698/107550338-2f401d80-6bd1-11eb-8b12-35f7e706a457.png)

**To Reproduce**
Deploy the example project onto an Android device similar to the one listed below.
The only changes made to the project: 
- updating `minSDKVersion` to 25 in build.gradle.
- Fixing build.cs to allow unity to export, like mentioned [here](https://github.com/juicycleff/flutter-unity-view-widget/issues/234#issuecomment-735307249)

The bug is not in version 0.1.6+5 and it is in version 3.0.0 and 3.0.3, I did not test any version in between those yet.
I tried the 4.0.0 alpha, but I haven't yet been able to run it without errors, on any phone.

**Desktop**
 - OS: Windows 10
 - Unity 2019.4.13f, exported with `export android`
 - flutter_windows_1.22.6-stable
 - Either `flutter_unity_widget: ^3.0.0` and `flutter_unity_widget: ^3.0.3`  in pubspec.yaml

**Smartphone**
 - Device: Wileyfox swift
 - OS: Android 7.1.2, sdk level 25
 - CPU: Qualcomm Snapdragon 410
 - GPU: Adreno 306 

**Additional context**
Both the outdated AR example and the alternative flutter package  `flutter_unity` show Unity normally on this device. That shows that it is capable of running it properly.

**Comment by tomascimadorook** [comment]
Same issue here, Samsung Galaxy S6 Android 7.0

**Comment by Altaf-jg** [comment]
This is happening on Android 6 devices as well. If we use old plugin, then few devices with new android OS (10+) has freeze issues & if we use old Android OS, we have rendering issues. This needs to be resolved soon for us. We are holding our release for this issue.

**Comment by juicycleff** [comment]
@timbotimbo, nice name. Please could you try out flutter_unity_widget: ^4.0.0-alpha.1 and see if it still exist?

**Comment by timbotimbo** [comment]
@juicycleff 
I can get the alpha to build and deploy, but it will always error when opening the unity view, on any phone.

Flutter throws an exception in `flutter\packages\flutter\lib\src\services\message_codecs.dart` in `dynamic decodeEnvelope` line 582.

```
java.lang.IllegalStateException: Trying to create a platform view of unregistered type: plugins.xraph.com/unity_view
	at io.flutter.plugin.platform.PlatformViewsController$1.createAndroidViewForPlatformView(PlatformViewsController.java:137)
	at io.flutter.embedding.engine.systemchannels.PlatformViewsChannel$1.create(PlatformViewsChannel.java:101)
	at io.flutter.embedding.engine.systemchannels.PlatformViewsChannel$1.onMethodCall(PlatformViewsChannel.java:59)
	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:233)
	at io.flutter.embedding.engine.dart.DartMessenger.handleMessageFromDart(DartMessenger.java:85)
	at io.flutter.embedding.engine.FlutterJNI.handlePlatformMessage(FlutterJNI.java:692)
	at android.os.MessageQueue.nativePollOnce(Native Method)
	at android.os.MessageQueue.next(MessageQueue.java:323)
	at android.os.Looper.loop(Looper.java:136)
	at android.app.ActivityThread.main(ActivityThread.java:6121)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
```

**Comment by avasich** [comment]
Same error here with v4 on all devices (which are xiaomi mi9t (android 11) and several emulators). Tried Unity 2020.1.9f1, 2020.1.17f1 and 2020.2.3f1. Before the error there is a warning in logcat that seems relevant:

`W/GeneratedPluginsRegister: Tried to automatically register plugins with FlutterEngine (io.flutter.embedding.engine.FlutterEngine@7e9a8e7) but could not find and invoke the GeneratedPluginRegistrant.`

Logging showed that GeneratedPluginRegistrant.registerWith() actually gets called, so I don't understand what's going on.

**Comment by juicycleff** [comment]
@timbotimbo @avasich, use the latest alpha. Contains a fix for Android and please update your FlutterUnitypackage with the latest release in scripts folder

**Comment by avasich** [comment]
@juicycleff thank you, it works! Still unable to build with 2020.2 for other reasons, but with 2020.1 everything's fine now.

**Comment by juicycleff** [comment]
@avasich I've not done any regression test against  2020.2, so there might be breaking change as it largely depends on the direction unity is taking

**Comment by mathtasatlime** [comment]
@juicycleff @timbotimbo at the mom, I'm facing exactly the same error of creating a view with an unregistered type:
`
[+5102 ms] [VERBOSE-2:ui_dart_state.cc(177)] Unhandled Exception: PlatformException(unregistered_view_type, trying to create a view with an unregistered type, unregistered view type:
'plugin.xraph.com/unity_view', null)
[   +3 ms] #0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:582:7)
[        ] #1      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:159:18)
[        ] <asynchronous suspension>
[        ] #2      MethodChannel.invokeMethod (package:flutter/src/services/platform_channel.dart:332:12)
[        ] #3      PlatformViewsService.initUiKitView (package:flutter/src/services/platform_views.dart:219:41)
[        ] #4      _UiKitViewState._createNewUiKitView (package:flutter/src/widgets/platform_view.dart:624:71)
[        ] #5      _UiKitViewState._initializeOnce (package:flutter/src/widgets/platform_view.dart:574:5)
[        ] #6      _UiKitViewState.didChangeDependencies (package:flutter/src/widgets/platform_view.dart:584:5)
[        ] #7      StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:4786:12)
[        ] #8      ComponentElement.mount (package:flutter/src/widgets/framework.dart:4601:5)
[        ] #9      Element.inflateWidget (package:flutter/src/widgets/framework.dart:3569:14)
[        ] #10     Element.updateChild (package:flutter/src/widgets/framework.dart:3327:18)
[        ] #11     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4652:16)
[  +14 ms] #12     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4800:11)
[        ] #13     Element.rebuild (package:flutter/src/widgets/framework.dart:4343:5)
[        ] #14     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:4606:5)
[        ] #15     StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:4791:11)
[        ] #16     ComponentElement.mount (package:flutter/src/widgets/framework.dart:4601:5)
[        ] #17     Element.inflateWidget (package:flutter/src/widgets/framework.dart:3569:14)
[        ] #18     Element.updateChild (package:flutter/src/widgets/framework.dart:3327:18)
[        ] #19     SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:6118:14)
[        ] #20     Element.inflateWidget (package:flutter/src/widgets/framework.dart:3569:14)
[        ] #21     Element.updateChild (package:flutter/src/widgets/framework.dart:3327:18)
[   +3 ms] #22     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4652:16)
[   +2 ms] #23     Element.rebuild (package:flutter/src/widgets/framework.dart:4343:5)
[        ] #24     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:4606:5)
[        ] #25     ComponentElement.mount (package:flutter/src/widgets/framework.dart:4601:5)
[        ] #26     Element.inflateWidget (package:flutter/src/widgets/framework.dart:3569:14)
[        ] #27     Element.updateChild (package:flutter/src/widgets/framework.dart:3327:18)
[        ] #28     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4652:16)
[        ] #29     Element.rebuild (package:flutter/src/widgets/framework.dart:4343:5)
[        ] #30     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:4606:5)
[        ] #31     ComponentElement.mount (package:flutter/src/widgets/framework.dart:4601:5)
[        ] #32     Element.inflateWidget (package:flutter/src/widgets/framework.dart:3569:14)
[        ] #33     Element.updateChild (package:flutter/src/widgets/framework.dart:3324:20)
[        ] #34     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4652:16)
[        ] #35     Element.rebuild (package:flutter/src/widgets/framework.dart:4343:5)
[        ] #36     StatelessElement.update (package:flutter/src/widgets/framework.dart:4708:5)
[        ] #37     Element.updateChild (package:flutter/src/widgets/framework.dart:3314:15)
[        ] #38     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4652:16)
[        ] #39     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4800:11)
[        ] #40     Element.rebuild (package:flutter/src/widgets/framework.dart:4343:5)
[        ] #41     BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2730:33)
[        ] #42     WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:913:20)
[        ] #43     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:302:5)
[        ] #44     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1117:15)
[        ] #45     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1055:9)
[        ] #46     SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:971:5)
[        ] #47     _rootRun (dart:async/zone.dart:1190:13)
[   +7 ms] #48     _CustomZone.run (dart:async/zone.dart:1093:19)
[        ] #49     _CustomZone.runGuarded (dart:async/zone.dart:997:7)
[        ] #50     _invoke (dart:ui/hooks.dart:251:10)
[        ] #51     _drawFrame (dart:ui/hooks.dart:209:3)
`

The error occurs directly when creating a unityView inside the flutter app. Im using v4.0.0-alpha 2 und also the up-to-date UnityPackage. I also deleted the iOS folder completely and started from scratch via the explanation in the readme of v4. Do you have an idea? @timbotimbo how did you solve it?

**Comment by juicycleff** [comment]
@mathtasatlime have you tried 'flutter clean'

**Comment by mathtasatlime** [comment]
@juicycleff unfortunately yes. Are you only using Xcode for testing the example project or to you code in vscode? 
building for android works fine, only the iOS building not. You changed the name for the view from "plugins.x..." to "plugin.x...". Maybe something about that?

**Comment by mathtasatlime** [comment]
@juicycleff the same error occurs in the example project. Im using Xcode 12 on a MacMini 2020

**Comment by mathtasatlime** [comment]
Okay, so I changed my project back to version 3.0.3 and it worked. I didn't get the errors again. @juicycleff if you need more information about this bug apperently still sitting in 4.0.0-alpha2, just let me now. I'm open to help :)

**Comment by avasich** [comment]
@juicycleff it turned out that ios version has the same error
`Unhandled Exception: PlatformException(unregistered_view_type, trying to create a view with an unregistered type, unregistered view type: 'plugin.xraph.com/unity_view', null)`

That's happening for the same reason as with bug you fixed last week for android. In SwiftFlutterUnityWidgetPlugin.swift and FLTUnityWidgetController.swift ids start with "plugins.xraph.com" instead of "plugin.xraph.com". With this lines fixed everything works.

**Comment by juicycleff** [comment]
@avasich thanks for this, I actually changed the method channel ID midway and didn't update iOS, made a new release containing this fix

**Comment by Altaf-jg** [comment]
Any update on this? We updated to  4.0.1+1 & still observe the same issue in Android 7 & below

**Comment by MaxymShtoyko** [comment]
@Altaf-jg @tomascimadorook @avasich @timbotimbo did you manage to find any solutions?

**Comment by Altaf-jg** [comment]
@MaxymShtoyko Not yet, still waiting for a solution. As a temporary solution we started distributing two APK`s one with new plugin & one with old plugin. Would be great if the issue get resolved in the proper way.

**Comment by guoguoguilai** [comment]
> @Altaf-jg @tomascimadorook @avasich @timbotimbo did you manage to find any solutions?
The same issue. After I change to version 4.0.1+1 and unity package 4.0, it disapears

**Comment by Altaf-jg** [comment]
@guoguoguilai Not yet, Its been three months since we reported the issue. No update from anyone yet.

**Comment by timbotimbo** [comment]
My only solution remains to be using this plugin for iOS and use the `flutter_unity` project for android.

**Comment by Anunth** [comment]
@juicycleff  unity urp materials are not showing in flutter 


Here is the log 

D/Unity   ( 6418):  GL_OES_EGL_image GL_OES_EGL_image_external GL_OES_EGL_sync GL_OES_vertex_half_float GL_OES_framebuffer_object GL_OES_rgb8_rgba8 GL_OES_compressed_ETC1_RGB8_texture GL_AMD_compressed_ATC_texture GL_KHR_texture_compression_astc_ldr GL_OES_texture_npot GL_EXT_texture_filter_anisotropic GL_EXT_texture_format_BGRA8888 GL_EXT_read_format_bgra GL_OES_texture_3D GL_EXT_color_buffer_float GL_EXT_color_buffer_half_float GL_QCOM_alpha_test GL_OES_depth24 GL_OES_packed_depth_stencil GL_OES_depth_texture GL_OES_depth_texture_cube_map GL_EXT_sRGB GL_OES_texture_float GL_OES_texture_float_linear GL_OES_texture_half_float GL_OES_texture_half_float_linear GL_EXT_texture_type_2_10_10_10_REV GL_EXT_texture_sRGB_decode GL_EXT_texture_format_sRGB_override GL_OES_element_index_uint GL_EXT_copy_image GL_EXT_geometry_shader GL_EXT_tessellation_shader GL_OES_texture_stencil8 GL_EXT_shader_io_blocks GL_OES_shader_image_atomic GL_OES_sample_variables GL_EXT_texture_border_clamp GL_EXT_EGL_image_external_wrap_modes GL_EXT_multisampled
D/Unity   ( 6418): _render_to_texture GL_EXT_multisampled_render_to_texture2 GL_OES_shader_multisample_interpolation GL_EXT_texture_cube_map_array GL_EXT_draw_buffers_indexed GL_EXT_gpu_shader5 GL_EXT_robustness GL_EXT_texture_buffer GL_EXT_shader_framebuffer_fetch GL_ARM_shader_framebuffer_fetch_depth_stencil GL_OES_texture_storage_multisample_2d_array GL_OES_sample_shading GL_OES_get_program_binary GL_EXT_debug_label GL_KHR_blend_equation_advanced GL_KHR_blend_equation_advanced_coherent GL_QCOM_tiled_rendering GL_ANDROID_extension_pack_es31a GL_EXT_primitive_bounding_box GL_OES_standard_derivatives GL_OES_vertex_array_object GL_EXT_disjoint_timer_query GL_KHR_debug GL_EXT_YUV_target GL_EXT_sRGB_write_control GL_EXT_texture_norm16 GL_EXT_discard_framebuffer GL_OES_surfaceless_context GL_OVR_multiview GL_OVR_multiview2 GL_EXT_texture_sRGB_R8 GL_KHR_no_error GL_EXT_debug_marker GL_OES_EGL_image_external_essl3 GL_OVR_multiview_multisampled_render_to_texture GL_EXT_buffer_storage GL_EXT_external_buffer GL_EXT_blit_framebuffer_par
D/Unity   ( 6418): ams GL_EXT_clip_cull_distance GL_EXT_protected_textures GL_EXT_shader_non_constant_global_initializers GL_QCOM_texture_foveated GL_QCOM_texture_foveated_subsampled_layout GL_QCOM_shader_framebuffer_fetch_noncoherent GL_QCOM_shader_framebuffer_fetch_rate GL_EXT_memory_object GL_EXT_memory_object_fd GL_EXT_EGL_image_array GL_NV_shader_noperspective_interpolation GL_KHR_robust_buffer_access_behavior GL_EXT_EGL_image_storage GL_EXT_blend_func_extended GL_EXT_clip_control GL_OES_texture_view GL_EXT_fragment_invocation_density GL_QCOM_validate_shader_binary GL_QCOM_YUV_texture_gather
I/SwappyDisplayManager( 6418): Starting looper thread
D/eu.meindoc.app( 6418): PlayerBase::PlayerBase()
D/eu.meindoc.app( 6418): TrackPlayerBase::TrackPlayerBase()
I/libOpenSLES( 6418): Emulating old channel mask behavior (ignoring positional mask 0x3, using default mask 0x3 based on channel count of 2)
W/AudioTrack( 6418): set(): notificationFrames=-21 clamped to the range -1 to -8
D/AudioTrack( 6418): createTrack_l(-1): AUDIO_OUTPUT_FLAG_FAST denied by server; frameCount 0 -> 4104
E/libc    ( 6418): Access denied finding property "ro.vendor.df.effect.conflict"
E/libc    ( 6418): Access denied finding property "ro.vendor.knock.type"
W/UnityMain( 6418): type=1400 audit(0.0:75905): avc: denied { read } for name="u:object_r:vendor_displayfeature_prop:s0" dev="tmpfs" ino=10935 scontext=u:r:untrusted_app:s0:c17,c257,c512,c768 tcontext=u:object_r:vendor_displayfeature_prop:s0 tclass=file permissive=0 app=eu.meindoc.app
V/MediaRouter( 6418): Selecting route: RouteInfo{ name=Phone, description=null, status=null, category=RouteCategory{ name=System types=ROUTE_TYPE_LIVE_AUDIO ROUTE_TYPE_LIVE_VIDEO  groupable=false }, supportedTypes=ROUTE_TYPE_LIVE_AUDIO ROUTE_TYPE_LIVE_VIDEO , presentationDisplay=null }

**Comment by obywan** [comment]
Ok, I have the same issue (camera output is rendered side by side with flutter app). Does anyone got this solved?

**Comment by folcjin** [comment]
https://folcjin.tistory.com/116

**Comment by Ahmadre** [comment]
I know what's the error eventually.

Try this solution: https://github.com/Ahmadre/unity_widget#solutions-to-problems

**Comment by timbotimbo** [comment]
Half of the comments above seem to be about unrelated issues.

@juicycleff 
I decided to revisit the original issue because I found an old Android 6 device.
I've used this plugin in production for a year without issues on iOS, but I still use a different plugin for Android because of this issue.

Tested with the project in the example folder.

Versions:
- Unity 2019.4.23  + Flutter_unity_widget 4.2.5+1
- Flutter 2.10.5

Devices:
- Samsung Galaxy J5, Android 6
- WileyFox Swift, Android 7

Both devices still have the original issue. Unitiy is rendered beneath the listview, and only partially visible when the device is in landscape mode.
![Screenshot_20170323-061546](https://user-images.githubusercontent.com/11444698/166442177-aa0c80b1-e812-4482-b44f-6ce8ef638de1.png)


I tried some UnityWidget parameter variations like fullscreen, AndroidViewSurface and uiLevel, but that didn't help.
```
UnityWidget(
    onUnityCreated: _onUnityCreated,
    onUnityCreated: _onUnityCreated,
    onUnityMessage: onUnityMessage,
    onUnitySceneLoaded: onUnitySceneLoaded,
    fullscreen: true,
    useAndroidViewSurface: true,
    uiLevel: 2,
),
```

Logcat of clicking 'Simple Unity Demo' on android 6.
```
2022-03-23 07:05:00.559 21225-21225/com.xraph.plugin.flutter_unity_widget_example D/ViewRootImpl: ViewPostImeInputStage processPointer 0
2022-03-23 07:05:00.629 21225-21225/com.xraph.plugin.flutter_unity_widget_example D/ViewRootImpl: ViewPostImeInputStage processPointer 1
2022-03-23 07:05:02.089 21225-21225/com.xraph.plugin.flutter_unity_widget_example I/Choreographer: Skipped 44 frames!  The application may be doing too much work on its main thread.
2022-03-23 07:05:02.449 21225-21253/com.xraph.plugin.flutter_unity_widget_example D/libGLESv1: DTS_GLAPI : DTS is not allowed for Package : com.xraph.plugin.flutter_unity_widget_example
2022-03-23 07:05:02.459 21225-21289/com.xraph.plugin.flutter_unity_widget_example I/Unity: MemoryManager: Using 'Dynamic Heap' Allocator.
2022-03-23 07:05:02.499 21225-21289/com.xraph.plugin.flutter_unity_widget_example I/Unity: SystemInfo CPU = ARM64 FP ASIMD AES, Cores = 4, Memory = 1383mb
2022-03-23 07:05:02.499 21225-21289/com.xraph.plugin.flutter_unity_widget_example I/Unity: SystemInfo ARM big.LITTLE configuration: 4 big (mask: 0xf), 0 little (mask: 0x0)
2022-03-23 07:05:02.499 21225-21289/com.xraph.plugin.flutter_unity_widget_example I/Unity: ApplicationInfo com.xraph.plugin.flutter_unity_widget_example version 1.0 build e69663eb-be7e-46fd-8631-a3b18be123ab
2022-03-23 07:05:02.509 21225-21289/com.xraph.plugin.flutter_unity_widget_example I/Unity: Built from '2019.4/staging' branch, Version '2019.4.23f1 (3f4e01f1a5ec)', Build type 'Release', Scripting Backend 'il2cpp', CPU 'armeabi-v7a', Stripping 'Enabled'
2022-03-23 07:05:02.639 21225-21289/com.xraph.plugin.flutter_unity_widget_example W/Adreno-EGL: <qeglDrvAPI_eglCreateContext:2469>: EGL_BAD_MATCH
2022-03-23 07:05:02.639 21225-21289/com.xraph.plugin.flutter_unity_widget_example W/Adreno-EGL: <qeglDrvAPI_eglGetConfigAttrib:607>: EGL_BAD_ATTRIBUTE
2022-03-23 07:05:02.639 21225-21289/com.xraph.plugin.flutter_unity_widget_example W/Adreno-EGL: <qeglDrvAPI_eglGetConfigAttrib:607>: EGL_BAD_ATTRIBUTE
2022-03-23 07:05:02.639 21225-21289/com.xraph.plugin.flutter_unity_widget_example W/Adreno-EGL: <qeglDrvAPI_eglGetConfigAttrib:607>: EGL_BAD_ATTRIBUTE
2022-03-23 07:05:02.669 21225-21289/com.xraph.plugin.flutter_unity_widget_example D/Unity:  GL_AMD_compressed_ATC_texture GL_AMD_performance_monitor GL_AMD_program_binary_Z400 GL_EXT_debug_label GL_EXT_debug_marker GL_EXT_discard_framebuffer GL_EXT_robustness GL_EXT_texture_format_BGRA8888 GL_EXT_texture_type_2_10_10_10_REV GL_NV_fence GL_OES_compressed_ETC1_RGB8_texture GL_OES_depth_texture GL_OES_depth24 GL_OES_EGL_image GL_OES_EGL_sync GL_OES_EGL_image_external GL_OES_element_index_uint GL_OES_fbo_render_mipmap GL_OES_fragment_precision_high GL_OES_get_program_binary GL_OES_packed_depth_stencil GL_OES_depth_texture_cube_map GL_OES_rgb8_rgba8 GL_OES_standard_derivatives GL_OES_texture_3D GL_OES_texture_float GL_OES_texture_half_float GL_OES_texture_half_float_linear GL_OES_texture_npot GL_OES_vertex_half_float GL_OES_vertex_type_10_10_10_2 GL_OES_vertex_array_object GL_QCOM_alpha_test GL_QCOM_binning_control GL_QCOM_driver_control GL_QCOM_perfmon_global_mode GL_QCOM_extended_get GL_QCOM_extended_get2 GL_QCOM_tiled_rendering GL_QCOM_writeonly_rendering GL_EXT_sRGB GL_EXT_sRGB_write_control GL_EXT
2022-03-23 07:05:02.669 21225-21289/com.xraph.plugin.flutter_unity_widget_example D/Unity: _texture_sRGB_decode GL_EXT_texture_filter_anisotropic GL_EXT_multisampled_render_to_texture GL_EXT_color_buffer_float GL_EXT_color_buffer_half_float GL_EXT_disjoint_timer_query

```

I'll try a new Unity install with 2022.1.0+2 soon, but I don't think much relevant code has changed for this issue.

**Comment by juicycleff** [comment]
Alright please test again, because I removed the zIndex and reset the orientation for the active activity and should fix it.

**Comment by timbotimbo** [comment]
@juicycleff 

I finally got a working version without this bug on android 6 & 7.
- Flutter 2.10.5
- flutter_unity_widget 2022.1.0+4 (and +3)
- Unity 2019.4.x 
- FlutterUnityIntegration-v4.1.0.unitypackage

Looks like the issue is fixed on package versions starting with 2022.1.

Unity 2019 with the unitypackage from 4.1 seems to be compatible with the latest 2022.1.0+4 package, and is easy to build.
Unity 2020 and 2022 had me solve gradle errors, compile errors and resulted in crashes anyway.
